spec/api/goals_api_spec.rb

```diff
@@ -28,4 +24,68 @@
       end
     end
   end
+
+  context 'GET#create' do
+
+    context 'with correct parameters' do
+      it 'creates a new goal as the current_user' do
+        user = create(:user)
+        headers = json_request_headers
+        headers['Authorization'] = ActionController::HttpAuthentication::Basic.encode_credentials(user.email,user.password)
+        params = { name: "the name of the goal" }
+
+        post '/api/goals/', params.to_json, headers
+        json = JSON.parse(response.body)
+
+        expect(response).to have_http_status(:ok)
+        expect(response.content_type).to eq('application/json')
+        expect(json).to eq({ "goal" => { "id" => 1, "name" => "the name of the goal", "created_at" => Time.now.strftime('%B %d %Y') }})
+      end
+    end
+
+    context 'with incorrect parameters' do
+      it 'returns an error' do
+        user = create(:user)
+        headers = json_request_headers
+        headers['Authorization'] = ActionController::HttpAuthentication::Basic.encode_credentials(user.email,user.password)
+        params = { title: "the name of the goal" }
+
+        post '/api/goals/', params.to_json, headers
+        json = JSON.parse(response.body)
+
+        expect(response).to have_http_status(:bad_request)
+        expect(json).to eq({ "error" => "Incorrect parameters" })
+      end
+    end
+  end
+
+  context 'PUT#update' do
+    it 'updates the goal with the correct parameters' do
+      user = create(:user)
+      goal = create(:goal, user: user, name: "the original name")
+      headers = json_request_headers
+      headers['Authorization'] = ActionController::HttpAuthentication::Basic.encode_credentials(user.email,user.password)
+      params = { name: "a new name that is different" }
+
+      put '/api/goals/1', params.to_json, headers
+      json = JSON.parse(response.body)
+
+      expect(response).to have_http_status(:ok)
+      expect(json["goal"]).to eq({ "id" => 1, "name" => "a new name that is different", "created_at" => Time.now.strftime('%B %d %Y') })
```

Comment:
@jeffminnear The id value isn't guaranteed to be `1`. How could the hardcoded id value `1` be avoided here?